hack/jenkins/e2e.sh

```diff
@@ -558,6 +558,9 @@ case ${JOB_NAME} in
     : ${GKE_API_ENDPOINT:="https://test-container.sandbox.googleapis.com/"}
     : ${E2E_CLUSTER_NAME:="gke-upgrade"}
     : ${E2E_NETWORK:="gke-upgrade"}
+    # TODO(ihmccreery): these really shouldn't be (very) version skewed, but
+    # because we have to rebuild, it could get slightly out of whack.
+    : ${E2E_OPT:="--check_version_skew=false"}
```

Comment:
I don't think that this is a good idea.  It will introduce difficult to diagnose flakes, as sometimes the client and server versions will be in sync, and sometimes not.  And sometimes the tests will fail, sometimes because of the version skew, etc, etc. 

Rather resolve the version skew between kubectl and master.  Do you know what's causing it?  There was a bug a while back, but I think it was resolved.  Let me dig it out for you.